File: apps/agent/app/collectors/system.py
import socket
import platform
import psutil

from app.models import SystemResponse

def collect_system() -> SystemResponse:
    physical_cores = psutil.cpu_count(logical=False)
    logical_cores = psutil.cpu_count(logical=True)
    memory = psutil.virtual_memory()
    boot_time = psutil.boot_time()

    return SystemResponse(
        hostname=socket.gethostname(),
        operating_system={
            "name": platform.system(),
            "release": platform.release(),
            "version": platform.version(),
            "architecture": platform.machine(),
        },
        cpu={
            "name": platform.processor() or "Unknown",
            "physical_cores": physical_cores or 0,
            "logical_cores": logical_cores or 0,
        },
        memory={
            "total_bytes": memory.total,
        },
        boot_time=boot_time,
    )

File: apps/web/i18n/en/system.ts
export const system = {
    connection: {
        label: "Connection state",

        states: {
            disconnected: "Disconnected",
            connecting: "Connecting",
            connected: "Connected",
            disconnecting: "Disconnecting",
            offline: "Offline",
            timed_out: "Timed out",
            error: "Error",
        },

        actions: {
            connect: "Connect agent",
            disconnect: "Disconnect",
            disconnecting: "Disconnecting...",
            cancel: "Cancel connection",
            retry: "Retry connection",
        },

        information: {
            service: "Service",
            status: "Status",
            version: "Version",
        },

        messages: {
            timedOut: "The BI Surface agent did not respond within 5 seconds.",
            offline: "The BI Surface agent could not be reached.",
            unknownError: "An unknown connection error occurred.",
            invalidHealth: "Agent returned an invalid health response.",
            invalidSystem: "Agent returned an invalid system response.",
            invalidTelemetry: "Agent returned an invalid telemetry response.",
            telemetryRefreshFailed: "Live telemetry could not be refreshed.",
            invalidProcesses: "Der Agent hat eine ungültige Prozessantwort zurückgegeben.",
            processesRefreshFailed: "Die Prozesssichtbarkeit konnte nicht aktualisiert werden.",
        },
    },

    overview: {
        hostname: "Hostname",
        operatingSystem: "Operating system",
        osVersion: "OS version",
        architecture: "Architecture",

        processor: "Processor",
        physicalCores: "Physical cores",
        logicalCores: "Logical cores",
        cpuUsage: "CPU usage",

        totalMemory: "Total memory",
        usedMemory: "Used memory",
        availableMemory: "Available memory",
        memoryUsage: "Memory usage",

        uptime: "Uptime",
    },
}
